pkg/storage/multi_sst_writer.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2023 The LevelDB-Go and Pebble Authors. All rights reserved. Use
+// of this source code is governed by a BSD-style license that can be found in
+// the LICENSE file.
+
+package storage
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/errors"
+	"github.com/cockroachdb/pebble"
+	"github.com/cockroachdb/pebble/internal/keyspan"
+	"github.com/cockroachdb/pebble/rangekey"
+	"github.com/cockroachdb/pebble/sstable"
+)
+
+// MultiSSTWriter may be used to construct a series of non-overlapping sstables
+// from a stream of keys. The user may specify a set of fixed boundaries such
+// that no sstable produced will span the boundaries.
```

Comment:
This comment was here previously not sure why it says "no sstable produced will span the boundaries", don't we want the opposite (i.e. to have SST's that span the `keyspans`)?